addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,408 +24,427 @@
+    if settings.timer then
+        if fiend == 'fiend' then
+            windower.send_command('timers c "'..current_mob..'" 30 down stun')
+            return new, color
+        end
+    end
+    if string.find(original,"remaining in Dynamis.") then
+        obj_time = (tonumber(original:match("%d+")) * 60)
+        end_time = os.time() + obj_time
+    end
+    if string.find(original,"will be expelled from Dynamis") then
+        obj_time = (tonumber(original:match("%d+")) * 60)
+        end_time = os.time() + obj_time
+    end
+    if string.find(original,"Your stay in Dynamis has been extended by %d+ minutes.") then
+        end_time = end_time + (tonumber(original:match("%d+")) * 60)
+        w.time = os.date('!%H:%M:%S', end_time)
+    end
+    item = original:match("Obtained key item: ..(%w+ %w+ %w+ %w+)..\46")
+    if item ~= nil then
+        item = item:lower()
+        for granule in Granules:it() do
+            if item == granule:lower() then
+                w[granule] = 1
+                init_window()
+            end
+        end
+    end
+    item = original:match("%w+ obtains an? ..(%w+ %w+ %w+ %w+)..\46")
+    --a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+ %w+ %w+)..\46")
+    if item == nil then
+        item = original:match("%w+ obtains an? ..(%w+ %w+ %w+)..\46")
+        --a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+ %w+)..\46")
+        if item == nil then
+            item = original:match("%w+ obtains an? ..(%w+%-%w+ %w+)..\46")
+            --a,b,item = string.find(original,"%w+ obtains an? ..(%w+%-%w+ %w+)..\46")
+            if item == nil then
+                item = original:match("%w+ obtains an? ..(%w+ %w+)..\46")
+                --a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+)..\46")
+            end
+        end
+    end
+    if item ~= nil then
+        item = item:lower()
+        for currency in Currencies:it() do
+            if item == currency:lower() then
+            w[currency] = w[currency] + 1
+            init_window()
+            end
+        end
+    end
+    return new, color
 end)
 
-function initializebox()
-	if obtained ~= nil and tracker == "on" then
- 		windower.text.create('dynamis_box')
- 		windower.text.set_bg_color('dynamis_box',200,30,30,30)
- 		windower.text.set_color('dynamis_box',255,200,200,200)
-		windower.text.set_location('dynamis_box',trposx,trposy)
- 		windower.text.set_visibility('dynamis_box',true)
- 		windower.text.set_bg_visibility('dynamis_box',true)
- 		windower.text.set_font('dynamis_box','Arial',12)
- 		windower.text.set_text('dynamis_box',obtained);
- 	end
-end
-
+-------------------------------------------------------------------------------
+-- Register target change event to get the monster name -----------------------
+-------------------------------------------------------------------------------
 windower.register_event('target change', function(targ_id)
```

Comment:
Also not your thing, but this variable name is wrong. It should be `targ_index` (or preferably `target_index`), it's not an ID.